src/main/java/de/szut/lf8_project/application/ProjectApplicationService.java

```diff
@@ -0,0 +1,112 @@
+package de.szut.lf8_project.application;
+
+import de.szut.lf8_project.common.*;
+import de.szut.lf8_project.controller.dtos.CreateProjectCommand;
+import de.szut.lf8_project.controller.dtos.ProjectView;
+import de.szut.lf8_project.domain.customer.CustomerService;
+import de.szut.lf8_project.domain.DateService;
+import de.szut.lf8_project.domain.adapter.EmployeeRepository;
+import de.szut.lf8_project.domain.customer.Customer;
+import de.szut.lf8_project.domain.customer.CustomerId;
+import de.szut.lf8_project.domain.employee.Employee;
+import de.szut.lf8_project.domain.employee.EmployeeId;
+import de.szut.lf8_project.domain.project.*;
+import de.szut.lf8_project.repository.RepositoryException;
+import de.szut.lf8_project.repository.projectRepository.ProjectRepository;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+
+@Service
+public class ProjectApplicationService {
+
+    private final EmployeeRepository employeeRepository;
+    private final ProjectRepository projectRepository;
+    private final DateService dateService;
+    private final CustomerService customerService;
+    private final ProjectService projectService;
+
+    public ProjectApplicationService(EmployeeRepository employeeRepository, ProjectRepository projectRepository, DateService dateService, CustomerService customerService, ProjectService projectService) {
+        this.employeeRepository = employeeRepository;
+        this.projectRepository = projectRepository;
+        this.dateService = dateService;
+        this.customerService = customerService;
+        this.projectService = projectService;
+    }
+
+    public ProjectView createProject(CreateProjectCommand cmd, JWT jwt) throws ApplicationServiceException {
+        validateProjectStartAndEnd(cmd.getStartDate(), cmd.getPlannedEndDate());
+        validateCustomer(cmd.getCustomerId());
```

Comment:
Dieses Design würd ich gerne hinterfragen.

Namentlich: wir haben void funktionen, wenn alles gut ist, passiert nichts, wenns nicht gut ist fliegen funken und exceptions. anstatt dass man zB bools zurückgibt.

ja, das beudetet eine extra if-schleife (irgendwo) - aber wenigstens kein try catch, code menge sollte also die gleiche bleiben
eigentlich ist das wieder ein usecase für eine `Resultat` Klase, ohne damit bewusst flashbacks triggern zu wollen.
das wir die hier nicht einführen ist sicherlich klar. suche grade noch nach einer anderen Lösung, bisher noch nichts gefunden...

Reply:
Ich bin einfach nicht so tief in der Materie wie ihr, but here goes:
- Hab keine Ahnung wie ein Either / Resultat aussieht. Das Argument für Exceptions war, dass wir die im Repo mit begrenzter Domänenlogik werfen können und dann nur noch durchreichen müssen. Wie wäre das mit Either/Resultat?
- Ich wäre dagegen im Repo eine Exception zu werfen und im Controller dann ein Resultat draus zu machen. Aber auch da könnt ihr mich gerne überstimmen.
- Wie viel zusätzliche Komplexität packen wir mit eurem Resultat / Either pattern mit rein? Ein Argument es nicht zu machen, wäre dass ich nicht noch ein Konzept verstehen muss.

- Ich hab auch nix dagegen die Excepitons einfach Runtime von Anfang an zu machen. dann knallen sie direkt bis zu SpringBoot durch :upside_down_face: (wenn ich das richtig verstanden habe) _In python ist sowieso alles runtime exceptions_ :man_shrugging: